src/main/scala/InsufficientMatingMaterial.scala

```diff
@@ -8,54 +8,86 @@ package chess
+  def nonKingPieces(board: Board) = nonKingPieceMap(board).toList
+  def boardWithNonKingPieces(board: Board) = board withPieces nonKingPieceMap(board)
+  def boardWithNonKingNonBishopPieces(board: Board) = board withPieces nonKingNonBishopPieceMap(board)
 
   /**
-   * Returns true when the only non-king pieces that remain are bishops that cannot
-   * capture each other and cannot checkmate (in atomic chess).
+   * Returns true when only the bishops are mobile and the bishops cannot capture.
+   * Assumes bishops can move through friendly pieces.
    */
-  def bishopsOnDifferentColor(board: Board) = {
+  def bishopsCannotCapture(board: Board) = noPieceMoves(boardWithNonKingNonBishopPieces(board)) && {
     val notKingPieces = nonKingPieces(board)
-    val onlyBishopsRemain = !notKingPieces.exists(_._2.role != Bishop)
+    val whitePlayerPieces = notKingPieces.filter {case (_, piece) => piece is Color.White}
+    val whitePlayerBishops = whitePlayerPieces.filter {case (_, piece) => piece is Bishop}
+    val blackPlayerPieces = notKingPieces.filter {case (_, piece) => piece is Color.Black}
+    val blackPlayerBishops = blackPlayerPieces.filter {case (_, piece) => piece is Bishop}
 
-    def bishopsOnSameColor  = notKingPieces.map(_._1.color).distinct.size == 1
-    def bishopsAreSameColor = notKingPieces.map(_._2.color).distinct.size == 1
+    !whitePlayerBishops.map {case (pos, _) => pos.color}.exists {
+      case color => blackPlayerPieces.exists {case (pos, _) => pos.color == color}
+    } && !blackPlayerBishops.map {case (pos, _) => pos.color}.exists {
+      case color => whitePlayerPieces.exists {case (pos, _) => pos.color == color}
+    }
+  }
 
-    if (!onlyBishopsRemain) false
-    else if (bishopsAreSameColor) notKingPieces.size < 3 || bishopsOnSameColor
-    else {
-      val whitePlayerBishops = notKingPieces.filter(_._2.color == Color.White)
-      val blackPlayerBishops = notKingPieces.filter(_._2.color == Color.Black)
+  /**
+   * Returns true when the only non-king pieces that remain are bishops that cannot checkmate.
+   */
+  def bishopsCannotCheckmate(board: Board) = {
+    val notKingPieces = nonKingPieces(board)
+    val onlyBishopsRemain = nonKingNonBishopPieceMap(board).isEmpty
+    lazy val piecesOnSameColor  = notKingPieces.map {case (pos, _) => pos.color}.distinct.size < 2
+    lazy val piecesAreSameColor = notKingPieces.map {case (_, piece) => piece.color}.distinct.size < 2
 
-      !whitePlayerBishops.exists {
-        case (pos, _) => blackPlayerBishops.exists(_._1.color == pos.color)
-      }
-    }
+    if (onlyBishopsRemain && piecesAreSameColor) piecesOnSameColor
+    else if (onlyBishopsRemain && !piecesOnSameColor) notKingPieces.size < 3
+    else bishopsCannotCapture(board)
   }
 
-  /*
+  /**
+   * Returns true when the only non-king pieces of a color that remain are bishops that cannot checkmate.
```

Comment:
Is this accurate?  this function calls bishopsCannotCapture, which looks at positions that include other pieces.